### ISSUE TITLE
New configuration system

### ISSUE THREAD

**Issue by ellisonleao** [ellisonleao/gruvbox.nvim#96]
Following #93, i am opening this issue to discuss a new configuration system for this plugin. The idea for the configuration is to have a table that can be overwritten and with some basic validation for default values.

@sQVe and others would love to hear some thoughts on that.

**Comment by sQVe** [comment]
@ellisonleao Sorry for not getting back at you sooner - I blame the holidays, two kids and a puppy 😅

I'll try to write down my point of view either tonight or tomorrow! I don't have any problems implementing things either, once we've actually made a decision.

**Comment by xeluxee** [comment]
I think [onedark](https://github.com/navarasu/onedark.nvim)'s configuration is well made and it could serve as an example.
It also allows users to customize [colors and highlights](https://github.com/navarasu/onedark.nvim#customization)

https://github.com/navarasu/onedark.nvim#default-configuration

**Comment by sQVe** [comment]
I agree and mirror @xeluxee opinion. Another example would be https://github.com/rebelot/kanagawa.nvim which has similar configuration options. The de facto standard `lua` way is using a `setup` function and I feel that we should only support that and drop the `g` variables.

Granular options like:

```lua
commentStyle = "italic",
functionStyle = "NONE",
keywordStyle = "italic",
statementStyle = "bold",
typeStyle = "NONE",
variablebuiltinStyle = "italic",
```
is really awesome, since it enables you to easily change the look/readability depending on fonts - without having to change any colors etc.

**Comment by ellisonleao** [comment]
New configuration system is now merged guys https://github.com/ellisonleao/gruvbox.nvim/releases/tag/1.0.0  . Please test and check if everything is working properly. Closing this issue
